### DISCUSSION TITLE
Help in setting up RDP for STM32H750B_DK

### DISCUSSION THREAD

**Discussion by MaazSk** [zephyrproject-rtos/zephyr#76297] [Q&A]
I tried some tests in zephyr/tests/drivers/flash and they worked. But I was unable to use zephyr WRP or RDP protection. I want to setup a RDP (readout protection) on my stm32h750b_dk but I am unable to set the same.

I am unable to set 
`CONFIG_FLASH_STM32_READOUT_PROTECTION=y` Even if I set I am unable to use `struct flash_stm32_ex_op_rdp` as it is used in the Extended operations api.

Is there a example to set Readout protection to `0xBB`. I am attaching a sample code that I was trying to use it is from my earlier discussion provided by @FRASTM. 
[flash_h7.zip](https://github.com/user-attachments/files/16378202/flash_h7.zip)

I am getting the following error
![image](https://github.com/user-attachments/assets/12a98adf-af36-4d85-b196-294aae70ed6c)

**Comment by FRASTM** [comment]
could this be an explaination : in the drivers/flash/Kconfig.stm32, the FLASH_STM32_READOUT_PROTECTION is not settable for the stm32H7 serie. 
```
config FLASH_STM32_READOUT_PROTECTION
	bool "Extended operation for flash readout protection control"
	depends on SOC_SERIES_STM32F4X || SOC_SERIES_STM32L4X || SOC_SERIES_STM32G4X
	select FLASH_HAS_EX_OP
```

There is a PR for adding the RDP to the stm32F7 serie : https://github.com/zephyrproject-rtos/zephyr/pull/76097
And I guess it probably requires one for the stm32H7 serie

**Reply by MaazSk** [reply]
Okay, I'll just check and if possible will create the PR, thank you! for the support.

**Reply by FRASTM** [reply]
> Okay, I'll just check and if possible will create the PR, thank you! for the support.

Maybe as an extension of the PR for F7 : does it make sens to add the stm32H7 serie at the same time ?

**Reply by MaazSk** [reply]
I think yes, just let me know if we can do that. I am not sure how to do it as an extension but will look into that as well

**Comment by FRASTM** [comment]
I put a PR https://github.com/zephyrproject-rtos/zephyr/pull/76640  to complete the https://github.com/zephyrproject-rtos/zephyr/pull/76097 

Can you please check if it answer your question ?

**Reply by MaazSk** [reply]
Sure will do that, thank you!

**Reply by MaazSk** [reply]
Hi @FRASTM,
I have not yet tested it but wanted to know when that PR will change it's state from Draft to complete?

**Reply by MaazSk** [reply]
Hi @FRASTM,
just wanted to know if we planning to merge the PR now or any other changes are also required?
Thank you!

**Comment by RajT33** [comment]
Hi @FRASTM ,
Just want to know in which release we will have this PR merged?
Thank you!
